FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt

```diff
@@ -655,15 +656,25 @@ class EmailUtil {
         pubKeys = mutableListOf()
         pubKeys.addAll(SecurityUtils.getRecipientsUsablePubKeys(context, recipients))
         pubKeys.addAll(SecurityUtils.getSenderPublicKeys(context, senderEmail))
-        prvKeys = listOf(
-          SecurityUtils.getSenderPgpKeyDetails(context, accountEntity, senderEmail).privateKey
-            ?: throw IllegalStateException("Sender private key not found")
-        )
-        ringProtector = KeysStorageImpl.getInstance(context).getSecretKeyRingProtector()
+
+        prvKeys = if (signingRequired) {
+          listOf(
+            SecurityUtils.getSenderPgpKeyDetails(context, accountEntity, senderEmail).privateKey
+              ?: throw IllegalStateException("Sender private key not found")
+          )
+        } else {
+          null
+        }
+
+        ringProtector = if (signingRequired) {
+          KeysStorageImpl.getInstance(context).getSecretKeyRingProtector()
+        } else {
+          null
+        }
```

Comment:
Maybe we can simplify this with below code?

```kt
if (signingRequired) {
  prvKeys = listOf(
    SecurityUtils.getSenderPgpKeyDetails(context, accountEntity, senderEmail).privateKey
      ?: throw IllegalStateException("Sender private key not found")
  )
  ringProtector = KeysStorageImpl.getInstance(context).getSecretKeyRingProtector() 
}
```

Reply:
Agree, that is more reasonable. `ringProtector` and `prvKeys` are probably initialized with `null` anyway?

Reply:
Thanks. It looks better. I will add changes.

Reply:
> Agree, that is more reasonable. `ringProtector` and `prvKeys` are probably initialized with `null` anyway?

Yes, you are right